### PR TITLE
HAAR-861: add disable external user

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt [hunk 1]
```diff
@@ -509,4 +509,81 @@ class UserControllerIntTest : IntegrationTestBase() {
...
+
+    @Test
+    fun `should fail with forbidden  for user not in group`() {
+      externalUsersApiMockServer.stubPutDisableFailUserNotInGroup()
+      webTestClient.put().uri("/users/2e285ccd-dcfd-4497-9e28-d6e8e10a2d3f/disable")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_OAUTH_USERS", "ROLE_AUTH_GROUP_MANAGER")))
+        .body(fromValue(mapOf("reason" to "bob")))
+        .exchange()
+        .expectStatus().isForbidden
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt:568]: Not sure if both tests are needed here.  Manage users is just the orchestrator so gets the success or failure.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt [hunk 2]
```diff
@@ -509,4 +509,81 @@ class UserControllerIntTest : IntegrationTestBase() {
...
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.put().uri("/users/2e285ccd-dcfd-4497-9e28-d6e8e10a2d3f/disable")
+        .headers(setAuthorisation(roles = listOf("ROLE_AUDIT")))
+        .body(fromValue(mapOf("reason" to "bob")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt:541]: Do we need to test with body missing?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesServiceTest.kt [hunk 3]
```diff
@@ -50,7 +50,16 @@ class RolesServiceTest {
...
+      whenever(
+        externalUsersService.getPagedRoles(
+          anyInt(),
+          anyInt(),
+          anyString(),
+          isNull(),
+          isNull(),
+          isNull()
+        )
+      ).thenReturn(roles)
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesServiceTest.kt:62]: Are you using a different ktLint to everybody else cos I'm sure mine didn't pick this up?  very odd

Comment (SimonMitchellMOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/RolesServiceTest.kt:62]: The was a couple of Assertions that weren't imported so ran `./gradlew ktlintFormat`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt (hunk 2 above)
Comment (SimonMitchellMOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserControllerIntTest.kt:541]: will add

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**